torch_xla/csrc/tensor.cpp

```diff
@@ -274,7 +274,13 @@ class XLATensor::DeviceContextArena {
 
  public:
   static DeviceContextArena* Get() {
-    static DeviceContextArena* arena = new DeviceContextArena();
+    static DeviceContextArena* arena = []() {
+      auto* ret = new DeviceContextArena();
+      // Setting a flag in ExecutionContext to alert that the device
+      // context is created and the execution mode cannot be changed now
+      ExecutionContext::Get()->set_device_context_created();
```

Comment:
we can use an env variable. The only con would be: It introduces another env variable for users to be aware of. If that's okay, I can change it to using an env variable instead of api call.

Reply:
I think we can keep it as an api for the following reasons:
1. One less env variable that the user needs to be aware of.
2. In future, if we want to provide the flexibility to switch between lazy and eager_debug in the middle of training runs, it can be made possible with this design.
3. It would be synonymous to the tensorflow 1.15 api for enable eager execution.
Let me know if this sounds right.

Reply:
I am fine with it being a API, but I don't find it useful to inject it to the DeviceContext. I can't think of a case where we only want to enable debug mode on a certain device. It could just be a process wide static boolean variable.

IMO it is a debugging flag we can enable, so I am hesitant to make it a mode(from the code structure pov). TF eager mode AFAIK has some reasonable performance and people can run their model eagerly. Eager debug mode this pr propsed is a purely for debugging. It shouldn't be in the same level as Lazy.

Reply:
The DeviceContextArena here is only for alerting the ExecutionContext class that the contxt arena has been created and any further attempt to change the execution mode will result in an error. This is just one mechanism to alert the user that they need to set the debug mode before the tensor creation.

Reply:
Isn't this with your vision of `provide the flexibility to switch between lazy and eager_debug in the middle of training runs`? If we want to guarantee this being set in the beginning, imo env var is a better approach.

Reply:
Yeah that is the vision and wanted to keep that door open. Since, we haven't felt the need to do that rn, kept a check to make sure that the setting of debug mode doesn't happen after tensor creation.
Creating an env variable would be the easiest way rn, but it would make it difficult to enable and disable the debug mode between training runs in future.
Moreover, it adds another env variable which I think might not be a great idea :)

Reply:
The vision was to have the switch sometime in the future. However, didn't have a strong reason yet to have that switch. Hence, was trying to make sure that the change doesn't happen after the tensors are registered. 
I agree the env variable would be a better and simpler approach if the purpose is to just set it at the beginning. Hence, have made the change to set a static variable that reads from an env variable.